src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -561,6 +563,9 @@ def create_or_upgrade_app(
                                 )
 
                         # hooks always executed after a create or upgrade
+                        get_cli_context().metrics.increment_counter(
+                            CLICounterField.POST_DEPLOY_SCRIPTS, 0
+                        )
```

Comment:
for the counting part, it's not a feature anymore. We would need to use something like counters (instead of features) for that, as it might mess with the feature graph.

Reply:
edit: count -> track, my bad. This is still a feature metric.

Reply:
Second comment discussed offline - we're separating tracking package scripts and post deploy hooks so that we can see the migration, so we added a new counter for that

First comment:
I tried removing the if statements from the respective calls and centralizing in `snowflake.cli.api.entities.utils.execute_post_deploy_hooks`, but there were some errors occurring in the specific `cls.use_package_warehouse` or `cls.use_application_warehouse` contextmanagers that you can find more details on [here](https://github.com/snowflakedb/snowflake-cli/actions/runs/10997800996/job/30534574111). It seems like it would be a sizeable refactor and I wasn't sure if it made sense in terms of separation of concerns to move the contextmanager over to the utils file.

I did a smaller refactor so that we only initialize the counters once in their respective files, let me know if this works for you!